### PR TITLE
common/gpu/intel/opencl: add opencl

### PR DESCRIPTION
**Pull request by Kiskae**
###### Description of changes

Updates intel libraries as per https://github.com/NixOS/nixpkgs/pull/205648
Adds module for intel opencl currently only supporting GPU

It is possible to add CPU OpenCL support, but only using an older version of the driver: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=intel-opencl-runtime

###### Things done

- [ ] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

**Comment by Kiskae** [comment]
The `intel-ocl` package is unfree and not required for newer processors, should it be gated behind an unfree/legacy option?

**Comment by Mic92** [comment]
> The `intel-ocl` package is unfree and not required for newer processors, should it be gated behind an unfree/legacy option?

How old? `hardware.enableRedistributableFirmware` could be used potentially used if the license allows for redistribution.

File: common/gpu/intel/opencl.nix [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{ config, lib, pkgs, ... }:
+
+{
+  options.hardware.intel.opencl.enable = lib.mkEnableOption (lib.mkDoc
+    "intel opencl runtime (Install icd for Intel GPUs)"
+  );
```
Comment (Mic92) [inline common/gpu/intel/opencl.nix:6]: No strong opinion here...:

```suggestion
  options.hardware.intel.opencl.enable = lib.mkEnableOption (lib.mkDoc
    "intel opencl runtime (Install icd for Intel GPUs)"
  ) // {
    enable = true;
  };
```

**Comment by Mic92** [comment]
Potentially makes sense to have the opencl option in nixos itself: https://github.com/NixOS/nixpkgs/pull/219657

File: common/gpu/intel/opencl.nix [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{ config, lib, pkgs, ... }:
+
+{
+  options.hardware.intel.opencl.enable = lib.mkEnableOption (lib.mkDoc
```
Comment (Mic92) [inline common/gpu/intel/opencl.nix:4]:
```suggestion
  options.hardware.intel.opencl.enable = lib.mkEnableOption (lib.mdDoc
```

**Comment by Kiskae** [comment]
> > The `intel-ocl` package is unfree and not required for newer processors, should it be gated behind an unfree/legacy option?
> 
> How old? `hardware.enableRedistributableFirmware` could be used potentially used if the license allows for redistribution.

Both the `intel-media-driver` and `intel-compute-runtime` support starts with Broadwell, aka 5th gen intel and newer. The [licence](https://www.intel.com/content/dam/support/us/en/documents/server-products/intel-eula-(distribution-only).pdf) does not include any provisions for redistribution.

In addition I realized that `intel-ocl` is a very old version of the CPU OpenCL driver. It also includes a GPU driver but that isn't actually packaged correctly.

File: common/gpu/intel/opencl.nix (hunk 2 above)
Comment (Pandapip1) [reply common/gpu/intel/opencl.nix:4]: Also, the module name is `hardware.intelgpu`, not `hardware.intel`.

**Pull request closed without merging**